src/shared/condition.c

```diff
@@ -555,9 +552,9 @@ static int condition_test_firmware(Condition *c, char **env) {
         assert(c->type == CONDITION_FIRMWARE);
 
         if (streq(c->parameter, "device-tree")) {
-                if (access("/sys/firmware/device-tree/", F_OK) < 0) {
+                if (access("/proc/device-tree/", F_OK) < 0) {
```

Comment:
Hmm, this chunk does a semantical change. Before, we would look for any of the device-tree trees, but now we only check for the base tree `/sys/firmware/device-tree/base`, a.k.a. `/proc/device-tree`. I don't think we want to do this, unless there's some additional justification.